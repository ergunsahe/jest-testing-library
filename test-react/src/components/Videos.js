import React from 'react'

const Videos = ({videos=[]}) => {

    if (videos.length <=0){
        return <h1>Sorry, no videos found</h1>
    }
    return (
        <div>
            {
                videos.map((video, index) =>{
                    return <h3 key={index} >{video}</h3>
                })
            }
        </div>
    )
}

export default Videos
