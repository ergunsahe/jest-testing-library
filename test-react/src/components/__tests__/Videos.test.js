import React from "react";
import renderer from "react-test-renderer";
import Videos from "../Videos"

test('Videos Empty Renders Sorry, no videos found', () =>{
    const tree = renderer.create(<Videos/>).toJSON()
    expect(tree).toMatchSnapshot()

})
test('Videos 1 Item Renders Correctly', () =>{
    const tree = renderer.create(<Videos videos={['udemy']}/>).toJSON()
    expect(tree).toMatchSnapshot()

})
test('Videos all Items Renders Correctly', () =>{
    const tree = renderer.create(<Videos videos={["udemy", "YouTube", "Tutorial"]}/>).toJSON()
    expect(tree).toMatchSnapshot()

})