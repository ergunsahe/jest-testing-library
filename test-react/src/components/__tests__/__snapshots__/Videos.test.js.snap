// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos 1 Item Renders Correctly 1`] = `
<div>
  <h3>
    udemy
  </h3>
</div>
`;

exports[`Videos Empty Renders Sorry, no videos found 1`] = `
<h1>
  Sorry, no videos found
</h1>
`;

exports[`Videos all Items Renders Correctly 1`] = `
<div>
  <h3>
    udemy
  </h3>
  <h3>
    YouTube
  </h3>
  <h3>
    Tutorial
  </h3>
</div>
`;
